src/devices_models/devices/thermal_generation.jl

```diff
@@ -517,6 +563,28 @@ function calculate_aux_variable_value!(
 
     return
 end
+
+function calculate_aux_variable_value!(
+    optimization_container::OptimizationContainer,
+    key::AuxVarKey{PowerOutput, T},
```

Comment:
we don't need the key here. Please use PowerOutput and T separately.

Reply:
we do use the key to get the aux container `aux_var_container = get_aux_variables(container)[key]`. This is just following the same template as TimeDurationOff/On

Reply:
we then need a method similar to: 

```Julia
function get_variable(
    container::OptimizationContainer,
    ::T,
    ::Type{U},
    meta::String = CONTAINER_KEY_EMPTY_META,
) where {T <: VariableType, U <: PSY.Component}
    return get_variable(container, VariableKey(T, U, meta))
end
```

for get_aux_variable